### PR TITLE
Train transfer learning models with different learning rate and epochs when fine-tuning

### PR DESCRIPTION
**Pull request by JulianZabbarov**
Adjusted model_to_transfer_learning_model function and pytorch model so that pytorch models can be fine-tuned using a different number of epochs and learning rates. Contributes to #8.

### CODE REVIEW

File: simba_ml/prediction/time_series/models/pytorch_lightning/dense_neural_network.py [hunk 1]
```diff
@@ -84,7 +84,18 @@ def forward(  # pylint: disable=arguments-differ
         )
 
     def configure_optimizers(self) -> torch.optim.Adam:
-        return torch.optim.Adam(self.parameters(), lr=1e-3)
+        if (
+            self.model_params.finetuning
+            and self.model_params.training_params.finetuning_learning_rate
```
[the remark this answers is not in the document]
Comment (JulianZabbarov) [reply simba_ml/prediction/time_series/models/pytorch_lightning/dense_neural_network.py:89]: When the user does not define the funetuning_learning_rate. This also satisfies mpyp so that finetuning_learning_rate is not None.

[the remark this answers is not in the document]
Comment (MaximilianKleissl) [reply simba_ml/prediction/time_series/models/pytorch_lightning/dense_neural_network.py:89]: Makes sense.

File: simba_ml/prediction/time_series/models/pytorch_lightning/pytorch_lightning_model.py [hunk 2]
```diff
@@ -144,11 +149,19 @@ def train(self, train: list[pd.DataFrame]) -> None:
             num_workers=0,
         )
 
-        trainer = pl.Trainer(
-            max_epochs=self.model_params.training_params.epochs,
-            log_every_n_steps=len(train_loader) // 2,
-            accelerator=self.model_params.training_params.accelerator,
-        )
+        if self.model_params.finetuning:
```
Comment (MaximilianKleissl) [inline simba_ml/prediction/time_series/models/pytorch_lightning/pytorch_lightning_model.py:152]: This method get's too long. Please refactor.

File: simba_ml/prediction/time_series/models/pytorch_lightning/dense_neural_network.py [hunk 3]
```diff
@@ -84,7 +84,19 @@ def forward(  # pylint: disable=arguments-differ
         )
 
     def configure_optimizers(self) -> torch.optim.Adam:
-        return torch.optim.Adam(self.parameters(), lr=1e-3)
+        if self.model_params.finetuning:
+            if self.model_params.training_params.finetuning_learning_rate is None:
+                raise ValueError(  # pragma: no cover
```
[the remark this answers is not in the document]
Comment (JulianZabbarov) [reply simba_ml/prediction/time_series/models/pytorch_lightning/dense_neural_network.py:89]: Oh, apparently you can access the hook! My sources said something different. 😓 Thanks for proving me wrong!

File: tests/prediction/time_series/models/pytorch_lightning_test.py [hunk 4]
```diff
@@ -24,11 +30,68 @@ def test_pytorch_lightning_dense_neural_network_predicts_correct_output_shape():
...
+        model.train(train=[train])
+
+
+def test_configure_optimizers_raises_error_if_finetuning_rate_not_set():
+    config = dense_neural_network.DenseNeuralNetworkConfig(finetuning=True)
+    time_series_params = time_series_config.TimeSeriesConfig(
+        input_features=[0, 1], output_features=[0, 1]
+    )
+    model = dense_neural_network.DenseNeuralNetwork(time_series_params, config)
+    logging.log(msg=str(model.model), level=logging.INFO)
```
Comment (MaximilianKleissl) [inline tests/prediction/time_series/models/pytorch_lightning_test.py:95]: Remove

**Review by MaximilianKleissl — APPROVED** [review]
LGTM
